x-pack/legacy/plugins/fleet/public/hooks/use_pagination.tsx

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useState } from 'react';
+
+const DEFAULT_PAGE_SIZE = 20;
+const PAGE_SIZES = [2, 20, 50];
```

Comment:
I'm not sure how useful this hook will be, but I don't mind trying it out. for agent list I pulled out these constants into [a common constants file](https://github.com/elastic/kibana/pull/47621/files#diff-5acb95a6abd0db803c1d81c064bc1300R15) so that they can be used on server-side as well (in case page size is not passed as part of the query, for example)

Reply:
Let's refactor to use the same thing after one of our PR is merged